online-fishing-store/src/app/pages/main-page/main-page.component.ts

```diff
@@ -25,18 +25,19 @@ export class MainPageComponent implements OnInit {
   constructor(
     public filterServise: FilterService,
     public sortServise: SortService,
-    public productServise: ProductService
+    public productService: ProductService
     ) {}
 
-  public getProducts(){
-    this.productServise.getProducts().subscribe((products) => {
-      this.products = products;
-      this.іnitialProducts = products;
-    })
+  public getProducts(): void{
+     this.productService.products$.subscribe((products:Product[]) => {
```

Comment:
We discussed about it yesterday.
You shouldn't use products$ subject in components.
You should use getProducts method from ProductService